tests/contrib/minikube_tests/test_kubernetes_pod_operator.py

```diff
@@ -0,0 +1,78 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import unittest
+from airflow.contrib.operators.kubernetes_pod_operator import KubernetesPodOperator
+from airflow import AirflowException
+from subprocess import check_call
+
+
+try:
+    check_call(["kubectl", "get", "pods"])
+except:
+    raise unittest.SkipTest(
+        "Kubernetes integration tests require a minikube cluster; Skipping tests"
+    )
+
+
+class KubernetesPodOperatorTest(unittest.TestCase):
+
+    def test_working_pod(self):
+        k = KubernetesPodOperator(namespace='default',
+                                  image="ubuntu:16.04",
+                                  cmds=["bash", "-cx"],
+                                  arguments=["echo", "10"],
+                                  labels={"foo": "bar"},
+                                  name="test",
+                                  task_id="task"
+                                  )
+
+        k.execute(None)
+
+    def test_faulty_image(self):
+        bad_image_name = "foobar"
```

Comment:
I guess the assumption here is that foobar will never be a valid image name in docker.io repository? Probably a safe assumption, but i'll call it out anyway 😄

Reply:
The image could also be in a domain that should never allow discovery of an image, ie. `example.com/bad/image`

Reply:
@ethernetdan should we limit what websites people can pull images from?

@cmkimerer fair point :), I can make it a bit more obscure to ensure it's never a hit (worst case it's a quick fix to change the image name for this test)